tests/ApplicationInsightsCompatibility/Intercept.Shared.Tests/TestEngine.cs

```diff
@@ -23,16 +23,14 @@ class TestEngine : TestEngineBase
 
         public const string MscorlibExtensionMethodsBaseModuleName = "Microsoft.Diagnostics.Instrumentation.Extensions.Base.dll";
         public const string InstrumentationEngineHostConfigName = "Microsoft.InstrumentationEngine.Extensions.config";
-        private const string InstrumentationEngineProfilerId = "{324F817A-7420-4E6D-B3C1-143FBED6D855}";
-        private const string InstrumentationEngineApmcExtensionApmcId = "{CA487940-57D2-10BF-11B2-A3AD5A13CBC0}";
-        private const string RawProfilerHookComponentId = "{4BCBE156-37EC-4179-B051-0183C57BACF9}";
 
         private TestEngine()
         {
         }
 
-        public static ITestResult ExecuteTest<T>()
+        public static ITestResult ExecuteTest<T>() where T : new()
         {
+            var _ = new T(); // instantiate the test class so the static analyzer doesn't think it's never used
```

Comment:
>var _ = new T(); // instantiate the test class so the static analyzer doesn't think it's never used [](start = 12, length = 99)

This is strange. I don't think I understand what the static analyzer is complaining about here. I don't think that we should be instantiating something here. It looks like a bug in the static analyzer. Can we disable the warning?

Reply:
This was the error: ``` tests/ApplicationInsightsCompatibility/Intercept.Shared.Tests/InterceptDecorateTests.cs(19,23): Warning CA1812: Avoid uninstantiated internal classes: InterceptDecorateTests.SimpleDecorate is an internal class that is apparently never instantiated. If so, remove the code from the assembly. If this class is intended to contain only static members, make it static (Shared in Visual Basic).```

I noticed I can make it go away just with the ```new()``` constraint, so I'll do that